radicle-httpd/src/api/test.rs

```diff
@@ -137,37 +137,69 @@ pub fn seed(dir: &Path) -> Context {
 pub async fn get(app: &Router, path: impl ToString) -> Response {
     Response(
         app.clone()
-            .oneshot(request(path, Method::GET, None))
+            .oneshot(request(path, Method::GET, None, None))
             .await
             .unwrap(),
     )
 }
 
-pub async fn post(app: &Router, path: impl ToString, body: Option<Body>) -> Response {
+pub async fn post(
+    app: &Router,
+    path: impl ToString,
+    body: Option<Body>,
+    auth: Option<String>,
+) -> Response {
     Response(
         app.clone()
-            .oneshot(request(path, Method::POST, body))
+            .oneshot(request(path, Method::POST, body, auth))
             .await
             .unwrap(),
     )
 }
 
-pub async fn put(app: &Router, path: impl ToString, body: Option<Body>) -> Response {
+pub async fn patch(
+    app: &Router,
+    path: impl ToString,
+    body: Option<Body>,
+    auth: Option<String>,
+) -> Response {
     Response(
         app.clone()
-            .oneshot(request(path, Method::PUT, body))
+            .oneshot(request(path, Method::PATCH, body, auth))
             .await
             .unwrap(),
     )
 }
 
-fn request(path: impl ToString, method: Method, body: Option<Body>) -> Request<Body> {
-    Request::builder()
+pub async fn put(
+    app: &Router,
+    path: impl ToString,
+    body: Option<Body>,
+    auth: Option<String>,
+) -> Response {
+    Response(
+        app.clone()
+            .oneshot(request(path, Method::PUT, body, auth))
+            .await
+            .unwrap(),
+    )
+}
+
+fn request(
```

Comment:
I didn't want to add the `Authorization` header to every request, so I went with a conditional adding, else we pass `None` to `request`